catalog/core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/DynamicSchemaResolverTest.java

```diff
@@ -97,6 +98,63 @@ public void testAddFields() throws Exception {
     }
   }
 
+  /**
+   * Verify that when the metadata size limit is set to 0 (less than the size of what is being added
+   * to the SolrInputDocument) that it is not added to the SolrInputDocument
+   */
+  @Test
+  public void testAddFieldsDoesntAddMetadataSizeLimitIsZero() throws Exception {
+    // Set
+    System.setProperty("metadata.size.limit", "0");
+
+    // Setup
+    String metacardTypeName = "states";
+    Set<AttributeDescriptor> attributeDescriptors = new HashSet<AttributeDescriptor>(1);
+    String propertyName = "title";
+    String name = metacardTypeName + "." + propertyName;
+    boolean indexed = true;
+    boolean stored = true;
+    boolean tokenized = false;
+    boolean multiValued = false;
+    attributeDescriptors.add(
+        new TestAttributeDescriptorImpl(
+            name, propertyName, indexed, stored, tokenized, multiValued, BasicTypes.OBJECT_TYPE));
+    Serializable mockValue = mock(Serializable.class);
+    Attribute mockAttribute = mock(Attribute.class);
+    when(mockAttribute.getValue()).thenReturn(mockValue);
+    Metacard mockMetacard = mock(Metacard.class, RETURNS_DEEP_STUBS);
+    when(mockMetacard.getMetacardType().getName()).thenReturn(metacardTypeName);
+    when(mockMetacard.getMetacardType().getAttributeDescriptors()).thenReturn(attributeDescriptors);
+    when(mockMetacard.getAttribute(name)).thenReturn(mockAttribute);
+    SolrInputDocument mockSolrInputDocument = mock(SolrInputDocument.class);
+    DynamicSchemaResolver resolver = new DynamicSchemaResolver();
+
+    // Perform Test
+    resolver.addFields(mockMetacard, mockSolrInputDocument);
+    Assert.assertNull(mockSolrInputDocument.getFieldValue("lux_xml"));
```

Comment:
❗️We should do this instead:
```suggestion
    assertThat(solrInputDocument.getFieldValue("lux_xml"), is(notNullValue()));
```

Reply:
fixed